core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/Controller.scala

```diff
@@ -555,7 +557,20 @@ class Controller(
                       10
                     )
                   }
-                case other => throw new AmberException("principal didn't return updated metadata")
+                case other =>
+                  eventListener.workflowExecutionErrorListener.apply(
+                    ErrorOccurred(
+                      Error(
+                        "principal didn't return updated metadata",
+                        "Engine:Controller:PrincipalInitialization",
+                        Map(
+                          "return_value" -> y.toString(),
```

Comment:
In `Error.convertToMap()`, we use camel case for the key of the map but here we use underscore. I suggest we always use camel case for the key.